File: airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/destination_async/AirbyteFileUtilsTest.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination_async;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

public class AirbyteFileUtilsTest {

  @Test
  void testByteCountToDisplaySize() {

    assertEquals("500 bytes", AirbyteFileUtils.byteCountToDisplaySize(500L));
    assertEquals("1.95 KB", AirbyteFileUtils.byteCountToDisplaySize(2000L));
    assertEquals("2.93 MB", AirbyteFileUtils.byteCountToDisplaySize(3072000L));
    assertEquals("2.67 GB", AirbyteFileUtils.byteCountToDisplaySize(2872000000L));
    assertEquals("1.82 TB", AirbyteFileUtils.byteCountToDisplaySize(2000000000000L));
  }

}

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/destination/async/AirbyteFileUtilsTest.kt
/*
 * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.integrations.destination.async

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test

class AirbyteFileUtilsTest {
    @Test
    internal fun testByteCountToDisplaySize() {
        Assertions.assertEquals("500 bytes", AirbyteFileUtils.byteCountToDisplaySize(500L))
        Assertions.assertEquals("1.95 KB", AirbyteFileUtils.byteCountToDisplaySize(2000L))
        Assertions.assertEquals("2.93 MB", AirbyteFileUtils.byteCountToDisplaySize(3072000L))
        Assertions.assertEquals("2.67 GB", AirbyteFileUtils.byteCountToDisplaySize(2872000000L))
        Assertions.assertEquals("1.82 TB", AirbyteFileUtils.byteCountToDisplaySize(2000000000000L))
    }
}
